R/Spectra.R

```diff
@@ -114,20 +564,206 @@ setValidity("Spectra", function(object) {
+                         class(object@backend), " [", date(), "]"))
+              object
+          })
+
+#' @rdname Spectra
+#'
+#' @importMethodsFrom S4Vectors merge
+#'
+#' @exportMethod merge
+setMethod("merge", c("Spectra", "Spectra"), function(x, y, ...) {
+    objs <- unname(c(x, y, ...))
+    pqs <- lapply(objs, function(z) z@processingQueue)
+    ## For now we stop if there is any of the processingQueues not empty. Later
+    ## we could even test if they are similar, and if so, merge.
+    if (any(lengths(pqs)))
+        stop("Can not merge Spectra objects with non-empty processing queue")
+    metad <- do.call(c, lapply(objs, function(z) z@metadata))
+    procs <- unique(unlist(lapply(objs, function(z) z@processing)))
+    object <- new(
+        "Spectra", metadata = metad,
+        backend = backendMerge(lapply(objs, function(z) z@backend)),
+        processing = c(procs, paste0("Merge ", length(objs),
+                                     " Spectra into one [", date(), "]"))
+    )
+    validObject(object)
+    object
+})
+
+#### ---------------------------------------------------------------------------
+##
+##                          ACCESSOR METHODS
+##
+#### ---------------------------------------------------------------------------
+
+#' @rdname Spectra
+setMethod("acquisitionNum", "Spectra", function(object)
+    acquisitionNum(object@backend))
+
+#' @rdname Spectra
+setMethod("centroided", "Spectra", function(object) {
+    centroided(object@backend)
+})
+
+#' @rdname Spectra
+setReplaceMethod("centroided", "Spectra", function(object, value) {
+    centroided(object@backend) <- value
+    object
+})
+
+#' @rdname Spectra
+setMethod("collisionEnergy", "Spectra", function(object) {
+    collisionEnergy(object@backend)
+})
+
+#' @rdname Spectra
+setReplaceMethod("collisionEnergy", "Spectra", function(object, value) {
+    collisionEnergy(object@backend) <- value
+    object
+})
+
+#' @rdname Spectra
+setMethod("fileNames", "Spectra", function(object) {
+    fileNames(object@backend)
+})
+
+#' @rdname Spectra
+setMethod("fromFile", "Spectra", function(object) fromFile(object@backend))
+
+#' @rdname Spectra
+setMethod("intensity", "Spectra", function(object, ...) {
+    NumericList(lapply(.peaksapply(object, ...), function(z) z[, 2]),
+                compress = FALSE)
+})
+
+#' @rdname Spectra
+setMethod("ionCount", "Spectra", function(object) {
+    if (length(object))
+        unlist(.peaksapply(object, FUN = function(pks, ...)
+            sum(pks[, 2], na.rm = TRUE)), use.names = FALSE)
+    else numeric()
+})
+
+#' @rdname Spectra
+setMethod("isCentroided", "Spectra", function(object, ...) {
+    if (length(object))
+        unlist(.peaksapply(object, FUN = .is_centroided_peaks),
+               use.names = FALSE)
+    else logical()
+})
+
+#' @rdname Spectra
+setMethod("isEmpty", "Spectra", function(x) {
+    if (length(x))
+        unlist(.peaksapply(x, FUN = function(pks, ...) nrow(pks) == 0),
```

Comment:
Because the `isEmpty(object@backend)` will read the full data. The `.peaksapply` runs the function (in parallel) separate for each file. So the memory demand is never higher than number of processes * list of peak matrices of a file.

Reply:
Ok, sounds reasonable. I guess caused by the large PR and the small pieces of code that were visible in the *diff view* I missed the whole picture here.